procdiscovery/pkg/process/process.go

```diff
@@ -3,19 +3,25 @@ package process
 import (
 	"bufio"
 	"fmt"
+	"github.com/odigos-io/odigos/common/envs"
 	"io"
 	"os"
 	"strings"
 
-	"github.com/odigos-io/odigos/common/envOverwrite"
+	"github.com/odigos-io/odigos/common/envs/envOverwrite"
 )
 
 type Details struct {
-	ProcessID int
-	ExeName   string
-	CmdLine   string
-	// Envs only contains the environment variables that we are interested in
-	Envs map[string]string
+	ProcessID    int
+	ExeName      string
+	CmdLine      string
+	Environments *ProcessEnvs
```

Comment:
Why is it a pointer? can it be `nil`?

Reply:
yup, you can see at getRelevantEnvVars that it can return nil for some situations